### PR TITLE
`collections.namedtuple`: add lookaside

### PR DESCRIPTION
**Pull request by nikitaved**
General jit support for namedtuple types that are being created inside jitted callables.

### CODE REVIEW

File: thunder/core/interpreter.py [hunk 1]
```diff
@@ -2588,6 +2593,70 @@ def impl(self, other):
...
+        defaults = wrap_const(defaults)
+
+    if module is None:
+        module = wrap_const(module)
+    # }
+
+    # Run opaque namedtuple {
+    @interpreter_needs_wrap
+    def create_namedtuple(typename: str, field_names: str, **kwargs):
+        namedtuple_type = collections.namedtuple(typename, field_names, **kwargs)
```
Comment (t-vi) [inline thunder/core/interpreter.py:2628]: If module is not in the kwargs it takes the module from the direct caller, i.e. it uses `thunder.core.interpreter`. Could you update providing the top stack frame's module if no module argument is given, please? 
```
In [1]:     from collections import namedtuple
   ...:     import thunder
   ...:     typename = "MyNamedTuple"
   ...:     field_names = ("a", "b", "c")
   ...: 
   ...:     # Test returnign just the type {
   ...:     def f():
   ...:         return namedtuple(typename, field_names)
   ...: 
   ...:     jf = thunder.jit(f)


In [2]: 

In [2]: jf()
Out[2]: thunder.core.interpreter.MyNamedTuple
```

Other than that, great work!

Comment (nikitaved) [reply thunder/core/interpreter.py:2632]: Addressed!

**Comment by nikitaved** [comment]
@t-vi , I have addressed the module issue, thank you for pointing it out! Would you have other comments?

**Review by t-vi — APPROVED** [review]
OK, seems good to me now. I need to fix the `tuple` vs. `tuple.__new__` by having a new lookaside, but this is the next step.

**Pull request merged**